packages/flutter_tools/gradle/src/main/groovy/flutter.groovy

```diff
@@ -731,6 +731,85 @@ class FlutterPlugin implements Plugin<Project> {
         }
     }
 
+    // Add a task that can be called on Flutter projects that prints application id of a build
+    // variant.
+    //
+    // This task prints the application id in this format:
+    //
+    // ApplicationId: com.example.my_id
+    //
+    // Format of the output of this task is used by `AndroidProject.getApplicationIdForVariant`.
+    private static void addTasksForPrintApplicationId(Project project) {
+        project.android.applicationVariants.all { variant ->
+            // Warning: The name of this task is used by `AndroidProject.getApplicationIdForVariant`.
+            project.tasks.register("print${variant.name.capitalize()}ApplicationId") {
```

Comment:
I think according to https://developer.android.com/reference/tools/gradle-api/7.4/com/android/build/api/variant/ApplicationVariant#applicationId() this depends on some task running that will merge the manifests. That means this task depends on another task (possibly as simple as assembleVariant). Given this I think you should add a dependsOn block as defined here. https://docs.gradle.org/current/userguide/tutorial_using_tasks.html#sec:task_dependencies

Reply:
I think you might actually want `variant.outputs.each` see example here https://developer.android.com/build/gradle-tips#configure-dynamic-version-codes to print the applicationId. 
But I am not sure how task registration interacts differently if you use outputs.

Reply:
FWIW I am trying to give good guidance here if you explore these things and they dont work you can ignore my comments.

Reply:
It looks like it is not depending on any task, I ran it after a clean, and it still works. Also the application id is per applicationvariant I think? i didn't see a property on variantoutput for applicationID.